### ISSUE TITLE
Document Viewer | Issue tags: associate an existing issue tag to doc via searchable dropdown

### ISSUE THREAD

**Issue by abbyraskin** [department-of-veterans-affairs/caseflow#1437]
**Story**
When looking at a document, I want to search for or select an existing issue tag so I can easily organize lengthy files.

## Acceptance Criteria

**Selecting an existing issue**
- Validate that when a user types in any existing letter, the options below narrow down to existing issues
- Validate that the options in the dropdown are tags created in any of the documents in the case.
- Validate an issue is added by clicking on the option in dropdown or clicking enter/return on keyboard
- Validate that the selected issue shows in the search bar
- Validate that if there are more than 4 options the dropdown is scrollable
- Validate that there is a gray vertical line on the left of dropdown arrow
- Validate that clicking right side of the vertical line will show options in dropdown and clicking on left side of vertical line will click into box to type in search
- Validate that there is a [remove icon](https://github.com/department-of-veterans-affairs/appeals-design-research/blob/master/Projects/Styleguide/Design%20files/Icons/Remove%20Icon.svg) on the right of issue tags
- Validate that the remove icon deletes the issue tag from the document
  - The issue tag is still present in the dropdown, unless it is the only existing tag of its kind across the application, in which case it gets deleted completely
- Issue tags should take up width of searchable dropdown. If it passes the width, it should extend to next row.
- Any issue tags associated with a particular document display in the Issue Tags column on that document's row on the Documents List page

## Specs

- Specs for gray vertical line 24px, 13px from dropdown arrow
- Specs for tag issues should follow document list, font size 14px, 8px padding around text, grey tag is 32px height, 3px radius for corners 

## Mocks

![image](https://cloud.githubusercontent.com/assets/13420618/24917275/a785d0d2-1eaa-11e7-97e5-39aa1a36161e.png)

![image](https://cloud.githubusercontent.com/assets/13420618/24917287/b00e4cde-1eaa-11e7-9fbc-e95e02a24b53.png)

When issue tags take up a the search box, it should extend another row (82 px height)

![image](https://cloud.githubusercontent.com/assets/13420618/24917299/b6bcdc30-1eaa-11e7-902b-c41716c1452b.png)

![saved tags](https://cloud.githubusercontent.com/assets/13420618/24666195/a08e4098-192d-11e7-816e-9ee7321f5de0.gif)
**Note: This is an older version something the interaction of adding new issue. Ignore the tag an issue copy inside the search-in menu.**

[Invision Prototype](https://projects.invisionapp.com/share/BVB8NSO49#/screens/228309104)

## Note
- Veteran ID was taken out of header. See [Invision Prototype](https://projects.invisionapp.com/share/BVB8NSO49#/screens/228309104)
 for updated visual 

- Search in dropdown not up to date yet. Mocks above are most up to date.

**Comment by gnakm** [comment]
@abbyraskin Please check the story and see if it needs love!

**Comment by abbyraskin** [comment]
@gnakm added a couple little notes...for the bullet that says "Validate that if there are more than 4 options" should the end of that phrase be  "the dropdown is scrollable"?

Also, if you delete a selected issue by clicking the x, the issue is removed from the selected issues list, but is not removed from the dropdown of options, correct? That would be my expectation, but now I'm wondering if we need a way to remove a saved issue (possibly not necessary for v1, as they could just not use an issue they added accidentally).

**Comment by abbyraskin** [comment]
Summarizing what we discussed in person -- we won't have the ability to remove saved issues from the dropdown itself in v1 -- doesn't seem very risky given that issues don't persist across cases, and that we'll eventually populate these automatically from VACOLS. (And we'll make sure this is noted in their pilot training)

**Comment by abbyraskin** [comment]
Note: break into two separate issues (creating tags, selecting existing)

**Comment by abbyraskin** [comment]
cleaning this up a bit to remove any AC alluding to adding new issue tags, as that functionality was previously broken out by #1515

**Comment by tejans24** [comment]
Since we don't have pre-populated tags, the tag has to be typed in and the dropdown won't show a list of options. Do we need to update the AC for this?

**Comment by abbyraskin** [comment]
They can only select an existing tag if the user has added it.

**Comment by astewarttistatech** [comment]
@tejans24  please view the current functionality of issue tags in reader
![readerflow](https://cloud.githubusercontent.com/assets/23038505/25619420/cb63424a-2f18-11e7-8072-1187e8e29761.gif)


Blocked by the ability to not create issue tags

**Comment by kierachell** [comment]
With multiple tags there seems to be .... **double scrollbars!**
<img width="435" alt="screen shot 2017-05-03 at 9 01 07 am" src="https://cloud.githubusercontent.com/assets/18075411/25661551/1fad212e-2fdf-11e7-8478-f4469ae8897d.png">

**Comment by gnakm** [comment]
That is a bug. @tejans24 can you fix the double scrolly?

**Comment by tejans24** [comment]
@gnakm yea, I'm taking a look at it.

**Comment by tejans24** [comment]
Double scrolling is fixed

**Comment by kierachell** [comment]
## Issue tags should take up width of searchable dropdown. If it passes the width, it should extend to next row.
![longissues](https://cloud.githubusercontent.com/assets/18075411/26086276/bdbf24d4-39b7-11e7-8aab-ae09f834db9b.gif)


sorry @tejans24 ... I had to...

**Comment by tejans24** [comment]
@kierachell There is a another issue; #1875 covers that.

**Comment by kierachell** [comment]
**PASSED**
made issue #2015 for follow up and to avoid scope creep
